runner/ansible/roles/2.2.8/tasks/main.yml

```diff
@@ -0,0 +1,31 @@
+---
+
+- name: Set Test ID
+  set_fact:
+    test_id: '2.2.8'
+
+- name: "{{ test_id }}.check"
+  shell: |
+    #  HANA and SPS versions are compatible
+    sid=$(sudo crm configure show | grep -m1 SID= | sed -e "s/.*SID=\(...\).*/\1/" | tr '[:upper:]' '[:lower:]')
+    full_version=$(sudo -i -u ${sid}adm HDB version | grep "version:" | sed -e "s/^.*:[\ ]*//")
+    hana_version=$(echo $full_version | cut -d. -f1)
+    sps_version=$(echo $full_version | cut -d. -f3)
+    sps_revision=$(echo $full_version | cut -d. -f4)
+    [[ "$hana_version" = "{{ expected[test_id] }}" ]] && exit 0
+    [[ "$hana_version" != "1" ]] && exit 1
```

Comment:
This `"$hana_version" != "1"` looks wrong. Actually, almost everyone will have hana version 2.xx.
This if condition is used to determine the next 2 if conditions, which I guess should have the values (`110` and `03` in the vars file)

Reply:
The section of code from the example this replicates is:
          # SAP HANA 2.0 is supported with all known SPS versions.
          if [[ "$hana_version" = "2" ]]; then echo passed; fi
          # For SAP HANA 1.0 you need version SPS10 rev3, SPS11 or newer.
          if [[ "$hana_version" = "1" ]]; then
            if [[ "$sps_version" -ge "110" ]]; then echo passed; fi
            if [[ "$sps_version" = "100" && "$sps_revision" = "03" ]]; then echo passed; fi
          fi
What should be happening is that if the hanna_version matches expected, then we exit with a pass. So, the test != 1 test only happens if the version is not 2. We could put 110 and 3 into the vars file, but if that won't ever change I am not sure we need to put it in the vars file.